### ISSUE TITLE
Add option to include license message on `ax` object

### ISSUE THREAD

**Issue by darribas** [darribas/contextily#25]
This should be the default (albeit an option) in the upcoming `add_basemap` method and will include a license message. The Stamen message can be seen [here](http://maps.stamen.com/#toner/12/37.7706/-122.3782):

- For Toner and Terrain: "Map tiles by Stamen Design, under CC BY 3.0. Data by OpenStreetMap, under ODbL".
- For Watercolor: Map tiles by Stamen Design, under CC BY 3.0. Data by OpenStreetMap, under CC BY SA.

**Comment by darribas** [comment]
For an example of how this could work:

```python
lic = ("Map tiles by Stamen Design, under CC BY 3.0. "\
               "Data by OpenStreetMap, under ODbL.")

plt.text(ext[0],ext[2], lic, size=8)
```

where `ext` is a list of the extent of the image (`[minX, minY, maxX, maxY]`).

**Comment by darribas** [comment]
In terms of the API, we could make a `license` option in `add_basemap` where the default is `"Map tiles by Stamen Design, under CC BY 3.0. Data by OpenStreetMap, under ODbL."`

**Comment by ljwolf** [comment]
should this mean that our `sources` are classes with more than one trait? 

say, a `namedtuple("TileSource", ['url','license','description'])`?

**Comment by darribas** [comment]
Addressed in #43, so closing for now. @ljwolf idea of coming up with a `Source` class is good and we should think about it for future releases. @ljwolf , feel free to open an issue with API details and we can discuss? This was way simpler in scope.
